### PR TITLE
Add sbt plugin (#60)

### PR DESCRIPTION
**Pull request by jonas**
Refs: #60

### CODE REVIEW

File: sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          sys.error("nativeBindgenHeader not configured")
+        },
+        nativeBindgenPath := None,
+        nativeBindgenPackage := None,
+        nativeBindgenExclude := None,
+        resourceDirectories in nativeBindgen := resourceDirectories.value,
+        sourceGenerators += Def.task { Seq(nativeBindgen.value) },
+        name in nativeBindgen := "ScalaNativeBindgen",
+        nativeBindgen := {
+          val output  = sourceManaged.value / "sbt-scala-native-bindgen" / "nativeBindgen.scala"
```
Comment (jonas) [inline sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala]: This assumes that the user does not want to commit the generated output. We should probably add a `nativeBindgenOutput` settings to make it configurable.

File: sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  object autoImport {
+    val nativeBindgenPath =
+      taskKey[Option[File]]("Path to the scala-native-bindgen executable")
+    val nativeBindgenHeader = taskKey[File]("C header file")
+    val nativeBindgenPackage =
+      settingKey[Option[String]]("Package for the generated code")
+    val nativeBindgenLink =
+      settingKey[Option[String]]("Name of library to be linked")
+    val nativeBindgenExclude = settingKey[Option[String]]("Exclude prefix")
+    val nativeBindgen        = taskKey[File]("Generate GraphQL API code")
```
Comment (kornilova-l) [inline sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala]: @Jonas, I suppose you forgot to change the description for `nativeBindgen`, I'll change it to "Generate bindings"

File: build.sbt [hunk 3]
```diff
@@ -91,19 +107,45 @@ lazy val samples = project
...
+    .settings(
+      crossSbtVersions := List(Versions.sbt013, Versions.sbt1),
+      scalaVersion := {
+        (pluginCrossBuild / sbtBinaryVersion).value match {
+          case "0.13" => Versions.scala210
+          case _      => Versions.scala212
+        }
+      }
+    )
+}
```
Comment (kornilova-l) [inline build.sbt]: What was the problem with Scala/sbt versions? How does this `project` function solve it?

Comment (jonas) [reply build.sbt]: The problem was that Scala Native requires `scalaVersion` 2.11 and sbt either 2.10 or 2.12. Except for the samples project which tests the  generated bindings, all other sbt projects use the `scalaVersion` associated with sbt.

File: sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala (hunk 2 above)
Comment (jonas) [reply sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala]: Good catch :+1: thanks

File: build.sbt (hunk 3 above)
Comment (kornilova-l) [reply build.sbt]: 👍

**Pull request merged**